### PR TITLE
Kernels for collinear spin

### PR DESCRIPTION
**Pull request by mfherbst**
Kernels for Hartree, power-nonlinearity and LDA functionals for collinear spin.

### CODE REVIEW

File: src/postprocess/chi0.jl [hunk 1]
```diff
@@ -126,7 +126,7 @@ sufficiently converged. By default the `self_consistent_field` routine of `DFTK`
 returns `3` extra bands, which are not converged by the eigensolver
 (see `n_ep_extra` parameter). These should be discarded before using this function.
 """
-@timing function apply_χ0(ham, ψ, εF, eigenvalues, δV::RealFourierArray;
+@timing function apply_χ0(ham, ψ, εF, eigenvalues, δV::RealFourierArray, δVβ=nothing;
```
Comment (antoine-levitt) [inline src/postprocess/chi0.jl:129]: that's pretty ugly, but we'll fix that with the array stuff hopefully

File: src/terms/terms.jl [hunk 2]
```diff
@@ -80,42 +80,67 @@ end
...
-    nothing  # by default, no kernel
-end
-
 function compute_kernel(basis::PlaneWaveBasis{T}; kwargs...) where {T}
-    k = zeros(T, prod(basis.fft_size), prod(basis.fft_size))
+    n_spin = basis.model.n_spin_components
+    kernel = zeros(T, n_spin * prod(basis.fft_size), n_spin * prod(basis.fft_size))
     for term in basis.terms
-        k .+= compute_kernel(term; kwargs...)
+        term == false && continue
```
Comment (antoine-levitt) [inline src/terms/terms.jl:106]: If we're not going to use the "strong zero" property, might as well return nothing there, no?

**Review by antoine-levitt — COMMENTED** [review]
Cool!

File: src/postprocess/chi0.jl (hunk 1 above)
Comment (mfherbst) [reply src/postprocess/chi0.jl:129]: Agree 100%. I just wanted to get it all to work first.

**Pull request merged**